doc/source/cluster/key-concepts.rst

```diff
@@ -0,0 +1,106 @@
+.. include:: we_are_hiring.rst
+
+.. _cluster-key-concepts:
+
+Key Concepts
+============
+
+Cluster
+-------
+
+A Ray cluster is a set of one or more nodes that are running Ray and share the
+same :ref:`head node<cluster-node-types>`.
+
+.. _cluster-node-types:
+
+Node types
+----------
+
+A Ray cluster consists of a :ref:`head node<cluster-head-node>` and a set of
+:ref:`worker nodes<cluster-worker-node>`.
+
+.. image:: ray-cluster.jpg
+    :align: center
+    :width: 600px
+
+.. _cluster-head-node:
+
+Head node
+~~~~~~~~~
+
+The head node is the first node started by the
+:ref:`Ray cluster launcher<cluster-launcher>` when trying to launch a Ray
+cluster. Among other things, the head node holds the :ref:`Global Control Store
+(GCS)<memory>` and runs the :ref:`autoscaler<cluster-autoscaler>`. Once the head
+node is started, it will be responsible for launching any additional
+:ref:`worker nodes<cluster-worker-node>`. The head node itself will also execute
+tasks and actors to utilize its capacity.
+
+.. _cluster-worker-node:
+
+Worker node
+~~~~~~~~~~~
+
+A worker node is any node in the Ray cluster that is not functioning as head node.
+Therefore, worker nodes are simply responsible for executing tasks and actors.
+When a worker node is launched, it will be given the address of the head node to
+form a cluster.
+
+.. _cluster-launcher:
+
+Cluster launcher
+----------------
+
+The cluster launcher is a process responsible for bootstrapping the Ray cluster
+by launching the :ref:`head node<cluster-head-node>`. For more information on how
+to use the cluster launcher, refer to
+:ref:`cluster launcher CLI commands documentation<cluster-commands>` and the
+corresponding :ref:`documentation for the configuration file<cluster-config>`.
+
+.. _cluster-autoscaler:
+
+Autoscaler
+----------
+
+The autoscaler is a process that runs on the :ref:`head node<cluster-head-node>`
+and is responsible for adding or removing :ref:`worker nodes<cluster-worker-node>`
+to meet the needs of the Ray workload while matching the specification in the
+:ref:`cluster config file<cluster-config>`. In particular, if the resource
+demands of the Ray workload exceed the current capacity of the cluster, the
+autoscaler will try to add nodes. Conversely, if a node is idle for long enough,
+the autoscaler will remove it from the cluster.
```

Comment:
Oh, actually found a spot where the autoscaler is described in greater detail:
https://docs.ray.io/en/latest/cluster/guide.html#autoscaling-with-ray 
Would that make sense to link here?

Reply:
Suggested addition:

To learn more about autoscaling, refer to the :ref:`Ray cluster deployment guide<deployment-guide-autoscaler>`.

(Can add a reference <deployment-guide-autoscaler> pointing to https://docs.ray.io/en/latest/cluster/guide.html#autoscaling-with-ray.)

Reply:
Need to add the reference.

Reply:
Added, I think it should be complete now